### ISSUE TITLE
`undIndent` is confusing and should be solved better

### ISSUE THREAD

**Issue by k4rtik** [LambdaQs/arrays#2]
There is also currently a warning about `undIdent_list`:

```
✗ make genqsharp
bnfc -mbnfc.mk --ocaml-menhir QSharp.cf && make all
223 rules accepted
[...]
menhir ParQSharp.mly
[...]
File "ParQSharp.mly", line 180, characters 0-13:
Warning: symbol undIdent_list is unreachable from any of the start symbol(s).
[...]
ocamllex LexQSharp.mll
60 states, 3329 transitions, table size 13676 bytes
ocamlc  -o TestQSharp BNFC_Util.ml AbsQSharp.ml SkelQSharp.ml ShowQSharp.ml PrintQSharp.ml ParQSharp.mli ParQSharp.ml LexQSharp.ml TestQSharp.ml
make[1]: Leaving directory '/Users/kartik/Hub/LambdaQs/arrays/src'
```

@jakezweifler you may want to take a look at this issue once.

**Comment by k4rtik** [comment]
Turns out that warning existed before the introduction of `undIdent`:
```
File "ParQSharp.mly", line 178, characters 0-10:
Warning: symbol ident_list is unreachable from any of the start symbol(s).
```

The fix is to comment the following line:
```
separator nonempty Ident ".";
```

**Comment by k4rtik** [comment]
Fixed in 835ac6ad65690b1138f5681607932dca3de10ae6
